lib/livebook/file_system/local.ex

```diff
@@ -28,162 +29,206 @@ defmodule Livebook.FileSystem.Local do
 
     FileSystem.Utils.assert_dir_path!(default_path)
 
-    %__MODULE__{default_path: default_path}
+    %__MODULE__{host_id: local_host_id(), default_path: default_path}
+  end
+
+  @doc false
+  def local_host_id() do
+    {:ok, hostname} = :inet.gethostname()
+    hostname
   end
```

Comment:
@josevalim let me know if this heuristic makes sense, or is it likely for different nodes to have the same hostname? I thought about using the list of ip addresses, but this is more likely to change while the application runs (?)

Reply:
I think in all of them you can use `node()` as the `host_id`?

Reply:
That's what I did initially, but this has two problems.

* more important: we build the local file system during configuration and at that point we may not be running in distributed mode yet (we could solve it by configuring the node during configuration, but this sort of dependency feels brittle to me)
* less important: theoretically different nodes can run on the same host

So I like using the hostname much more, but ideally it would be something even more unique.

Reply:
I went with the pid idea and tested it works with two local Livebook nodes:

```elixir
iex(l@mockingjay)17> Livebook.FileSystem.File.list(file) |> elem(0)
:ok
iex(l@mockingjay)18> Node.spawn(:r@mockingjay, fn -> Livebook.FileSystem.File.list(file) |> IO.inspect() end)
#PID<23570.417.0>
{:error, "this local file system belongs to a different host"}
```

Reply:
Let me know if you have any idea to write a test. We would need a pid associated with another node, but something like `node(IEx.Helpers.pid(23570, 417, 0))` doesn't work without the actual node.

```elixir
  test "file operations an error when the file system is on a different host" do
    file_system = %{Local.new() | pid: OTHER_NODE_PID}
    dir_path = "/"

    assert {:error, "this local file system belongs to a different host"} =
             FileSystem.list(file_system, dir_path, false)
  end
```

Reply:
I don't think this can be easily tested, unfortunately. But since it is an error case, I would say it is fine as is.